pkg/ovn/types/command.go

```diff
@@ -0,0 +1,209 @@
+package types
+
+import (
+	k8stypes "github.com/ovn-kubernetes/ovn-kubernetes-mcp/pkg/kubernetes/types"
+)
+
+// Database represents an OVN database type.
+type Database string
+
+const (
+	// NorthboundDB is the OVN Northbound database.
+	NorthboundDB Database = "nbdb"
+	// SouthboundDB is the OVN Southbound database.
+	SouthboundDB Database = "sbdb"
+)
+
+// ShowParams are the parameters for ovn-nbctl/ovn-sbctl show command.
+type ShowParams struct {
+	k8stypes.NamespacedNameParams
+	Database Database `json:"database"`
+	MaxLines int      `json:"max_lines,omitempty"`
+}
+
+// ShowResult contains the output of ovn-nbctl/ovn-sbctl show command.
+type ShowResult struct {
+	Database Database `json:"database"`
+	Output   string   `json:"output"`
+}
+
+// ListTableParams are the parameters for listing records in an OVN table.
+type ListTableParams struct {
+	k8stypes.NamespacedNameParams
+	Database Database `json:"database"`
+	Table    string   `json:"table"`
+	Columns  string   `json:"columns,omitempty"`
+	Filter   string   `json:"filter,omitempty"`
+	MaxLines int      `json:"max_lines,omitempty"`
+}
+
+// ListTableResult contains records from an OVN table.
+type ListTableResult struct {
+	Database Database `json:"database"`
+	Table    string   `json:"table"`
+	Records  []string `json:"records"`
+}
+
+// LogicalSwitchListParams are the parameters for listing logical switches.
+type LogicalSwitchListParams struct {
+	k8stypes.NamespacedNameParams
+	Filter   string `json:"filter,omitempty"`
+	MaxLines int    `json:"max_lines,omitempty"`
+}
+
+// LogicalSwitchListResult contains the list of logical switches.
+type LogicalSwitchListResult struct {
+	Switches []string `json:"switches"`
+}
+
+// LogicalRouterListParams are the parameters for listing logical routers.
+type LogicalRouterListParams struct {
+	k8stypes.NamespacedNameParams
+	Filter   string `json:"filter,omitempty"`
+	MaxLines int    `json:"max_lines,omitempty"`
+}
+
+// LogicalRouterListResult contains the list of logical routers.
+type LogicalRouterListResult struct {
+	Routers []string `json:"routers"`
+}
+
+// LogicalSwitchPortListParams are the parameters for listing logical switch ports.
+type LogicalSwitchPortListParams struct {
+	k8stypes.NamespacedNameParams
+	Switch   string `json:"switch,omitempty"`
+	Filter   string `json:"filter,omitempty"`
+	MaxLines int    `json:"max_lines,omitempty"`
+}
+
+// LogicalSwitchPortListResult contains the list of logical switch ports.
+type LogicalSwitchPortListResult struct {
+	Switch string   `json:"switch,omitempty"`
+	Ports  []string `json:"ports"`
+}
+
+// ACLListParams are the parameters for listing ACLs.
+type ACLListParams struct {
+	k8stypes.NamespacedNameParams
+	Entity   string `json:"entity,omitempty"` // Logical switch or port group name
```

Comment:
Since logical switch or port group name is required as the value of the `Entity` field, the `omitempty` option need not be added.